src/Microsoft.Azure.WebJobs.Host/Config/FluentBindingRule.cs

```diff
@@ -122,9 +161,8 @@ internal FluentBindingRule<TAttribute> SetPostResolveHook(Func<TAttribute, Param
         /// <returns></returns>
         public void BindToInput<TType>(IConverter<TAttribute, TType> builderInstance)
         {
-            var bf = _parent.BindingFactory;
-            var rule = bf.BindToInput<TAttribute, TType>(builderInstance);
-            Bind(rule);
+            var pm = PatternMatcher.New(builderInstance);
```

Comment:
>PatternMatcher.New [](start = 21, length = 18)

I used PatternMatcher helpers to sanitize the various overloads.
Also, C# can't overload Func<TSrc, TDest> vs. Func<TSrc, Task<TDest>> 
But if we toss in a cancelation token, that changes the signature and then you can get the task overloads.